crates/factor-outbound-http/src/wasi.rs

```diff
@@ -61,10 +70,261 @@ impl<'a> WasiHttpView for WasiHttpImplInner<'a> {
 
     fn send_request(
         &mut self,
-        _request: Request<wasmtime_wasi_http::body::HyperOutgoingBody>,
-        _config: wasmtime_wasi_http::types::OutgoingRequestConfig,
+        mut request: Request<wasmtime_wasi_http::body::HyperOutgoingBody>,
+        mut config: wasmtime_wasi_http::types::OutgoingRequestConfig,
     ) -> wasmtime_wasi_http::HttpResult<wasmtime_wasi_http::types::HostFutureIncomingResponse> {
-        // TODO: port implementation from spin-trigger-http
-        Err(ErrorCode::HttpRequestDenied.into())
+        if let Some(interceptor) = &self.state.request_interceptor {
+            let intercepted = Intercepted {
+                request: &mut request,
+                config: &mut config,
+            };
+            if let Some(res) = interceptor.intercept(intercepted) {
+                return res;
+            }
+        }
+
+        let host = request.uri().host().unwrap_or_default();
+        let tls_client_config = self
+            .state
+            .component_tls_configs
+            .get_client_config(host)
+            .clone();
+
+        Ok(HostFutureIncomingResponse::Pending(
+            wasmtime_wasi::runtime::spawn(
+                send_request_impl(
+                    request,
+                    config,
+                    self.state.allowed_hosts.clone(),
+                    tls_client_config,
+                )
+                .in_current_span(),
+            ),
+        ))
     }
 }
+
+async fn send_request_impl(
+    mut request: Request<wasmtime_wasi_http::body::HyperOutgoingBody>,
+    mut config: wasmtime_wasi_http::types::OutgoingRequestConfig,
+    allowed_hosts: OutboundAllowedHosts,
+    tls_client_config: Arc<ClientConfig>,
+) -> anyhow::Result<Result<IncomingResponse, ErrorCode>> {
+    let allowed_hosts = allowed_hosts.resolve().await?;
+
+    let is_relative_url = request.uri().authority().is_none();
```

Comment:
This skips a lot of logic from the current impl but afaict `Authority` enforces everything we care about.